lib/twilio/device.ts

```diff
@@ -559,22 +557,45 @@ class Device extends EventEmitter {
    * @param options
    */
   async connect(options: Device.ConnectOptions = { }): Promise<Call> {
-    this._log.debug('.connect', JSON.stringify(options && options.params || {}), options);
+    this._log.debug('.connect', JSON.stringify(options));
     this._throwIfDestroyed();
-
     if (this._activeCall) {
       throw new InvalidStateError('A Call is already active');
     }
 
-    const activeCall = this._activeCall = await this._makeCall(
-      options.params || { },
-      {
-        enableImprovedSignalingErrorPrecision:
-          !!this._options.enableImprovedSignalingErrorPrecision,
-        rtcConfiguration: options.rtcConfiguration,
-        voiceEventSidGenerator: this._options.voiceEventSidGenerator,
-      },
-    );
+    let callSid, customParameters, parameters, signalingReconnectToken;
+    if (options.connectToken) {
+      try {
+        const parts = JSON.parse(atob(options.connectToken));
```

Comment:
`parts` seems too general of a variable name, how about `connectTokenParts` or `connectTokenParams`?